src/ui/hooks/useNetworkDevices.tsx

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useEffect, useState } from 'react';
+import {
+  MulticastDnsEventType,
+  MulticastDnsInformation,
+  useAvailableMulticastDnsDevicesListQuery,
+  useMulticastDnsMonitorUpdatesSubscription,
+} from '../gql/generated/types';
+import client from '../gql';
+
+export default function useNetworkDevices() {
+  const [networkDevices, setNetworkDevices] = useState<{
+    [name: string]: MulticastDnsInformation;
+  }>({});
+
+  const {
+    loading: multicastDnsDevicesListLoading,
+    data: multicastDnsDevicesListData,
+    error: multicastDnsDevicesListError,
+  } = useAvailableMulticastDnsDevicesListQuery({
+    fetchPolicy: 'network-only',
+    client,
+  });
+
+  useEffect(() => {
+    if (multicastDnsDevicesListData) {
+      const multicastDnsDevicesCopy = { ...networkDevices };
+      multicastDnsDevicesListData.availableMulticastDnsDevicesList.forEach(
+        (item) => {
+          multicastDnsDevicesCopy[item.name] = item;
+        }
+      );
+      setNetworkDevices(multicastDnsDevicesCopy);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [multicastDnsDevicesListData]);
+
+  const [newNetworkDevices, setNewNetworkDevices] = useState<
+    MulticastDnsInformation[]
+  >([]);
+
+  useMulticastDnsMonitorUpdatesSubscription({
+    fetchPolicy: 'network-only',
+    client,
+    onSubscriptionData: (options) => {
+      const data = options.subscriptionData.data?.multicastDnsMonitorUpdates;
+      if (data) {
+        const multicastDnsDevicesCopy = { ...networkDevices };
+
+        if (data?.type === MulticastDnsEventType.DeviceAdded) {
+          multicastDnsDevicesCopy[data.data.name] = data.data;
+
+          const newDevices = newNetworkDevices.map((item) => item);
+          newDevices.push(data.data);
+          setNewNetworkDevices(newDevices);
+        } else if (data?.type === MulticastDnsEventType.DeviceUpdated) {
+          multicastDnsDevicesCopy[data.data.name] = data.data;
+        } else if (data?.type === MulticastDnsEventType.DeviceRemoved) {
+          delete multicastDnsDevicesCopy[data.data.name];
+        }
+
+        setNetworkDevices(multicastDnsDevicesCopy);
+      }
+    },
+  });
+
+  const removeDeviceFromNewList = useCallback(
+    (deviceName: string) => {
+      setNewNetworkDevices(
+        newNetworkDevices.filter((d) => d.name !== deviceName)
```

Comment:
Yes, device name should always be unique, it is comprised of "elrs_<devicetype>_<mac address>", and since mac address will be unique the device name will be unique.

Reply:
Awesome!